graphql_compiler/tests/test_compiler.py

```diff
@@ -5542,7 +5578,25 @@ def test_filter_count_with_runtime_parameter_in_fold_scope(self):
         '''
         expected_gremlin = NotImplementedError
 
-        expected_sql = NotImplementedError
+        expected_sql = '''
+            SELECT
+                coalesce(folded_subquery_1.fold_output_name, ARRAY[]::VARCHAR[]) AS child_names,
+                [Animal_1].name AS name
+            FROM db_1.schema_1.[Animal] AS [Animal_1]
+            LEFT OUTER JOIN (
+                SELECT
+                    [Animal_2].uuid AS uuid,
+                    coalesce(count(*), 0) AS fold_output_x_count,
+                    array_agg([Animal_3].name) AS fold_output_name
+                FROM db_1.schema_1.[Animal] AS [Animal_2]
+                JOIN db_1.schema_1.[Animal] AS [Animal_3]
+                ON [Animal_2].uuid = [Animal_3].parent
+                GROUP BY [Animal_2].uuid
+            ) AS folded_subquery_1
```

Comment:
Nice work, these queries already look like something I wouldn't want to write by hand :)